components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -356,6 +356,8 @@ class GeckoEngineSession(
             if (initialLoad && url == ABOUT_BLANK) {
                 return
             }
+
+            currentUrl = url
```

Comment:
Looks like we only use the `currentUrl` in `onLocationChange` and `onTitleChange`. Did you verify this still works correctly (in the right order). :)

Reply:
Didn't think about the order.  Just tested and they are in the right order.
Before change: `onPageStart` -> `onTitleChange`
After change: `onLocationChange` -> `onTitleChange`